ksqldb-engine/src/main/java/io/confluent/ksql/internal/UtilizationMetricsListener.java

```diff
@@ -0,0 +1,193 @@
+        try {
+            final Collection<? extends Metric> allMetrics = streams.metrics().values();
+            final Map<String, Double> consumerMetrics = allMetrics.stream()
+                .filter(m -> m.metricName().group().equals(CONSUMER_METRICS)
+                    && metrics.contains(m.metricName().name())
+                    && !m.metricName().tags().getOrDefault("client-id", "").contains("global-consumer")
+                    && !m.metricName().tags().getOrDefault("client-id", "").contains("restore-consumer")
+                    && m.metricName().tags().getOrDefault("client-id", "").contains(threadName))
+                .collect(Collectors.toMap(k -> k.metricName().name(), v -> (Double) v.metricValue()));
+
+            final Map<String, Double> restoreConsumerMetrics = allMetrics.stream()
+                .filter(m -> m.metricName().group().equals(CONSUMER_METRICS)
+                    && metrics.contains(m.metricName().name())
+                    && !m.metricName().tags().getOrDefault("client-id", "").contains("global-consumer")
+                    && m.metricName().tags().getOrDefault("client-id", "").contains("restore-consumer")
+                    && m.metricName().tags().getOrDefault("client-id", "").contains(threadName))
+                .collect(Collectors.toMap(k -> k.metricName().name(), v -> (Double) v.metricValue()));
+
+            final Map<String, Double> producerMetrics = allMetrics.stream()
+                .filter(m -> m.metricName().group().equals(PRODUCER_METRICS)
+                    && metrics.contains(m.metricName().name())
+                    && m.metricName().tags().getOrDefault("client-id", "").contains(threadName))
+                .collect(Collectors.toMap(k -> k.metricName().name(), v -> (Double) v.metricValue()));
+
+            final Map<String, Double> threadMetrics = allMetrics.stream()
+                    .filter(m -> m.metricName().group().equals(STREAM_METRICS) &&
+                            m.metricName().tags().getOrDefault(THREAD_ID, "").equals(threadName) &&
+                            metrics.contains(m.metricName().name()))
+                    .collect(Collectors.toMap(k -> k.metricName().name(), v -> (Double) v.metricValue()));
+
+            final List<Metric> startTimeList = allMetrics.stream()
+                    .filter(m -> m.metricName().name().equals("thread-start-time") &&
+                            m.metricName().tags().getOrDefault(THREAD_ID, "").equals(threadName))
+                    .collect(Collectors.toList());
+            final Long threadStartTime = startTimeList.size() != 0 ? (Long) startTimeList.get(0).metricValue() : 0L;
+
+            double blockedTime = 0.0;
+            if (threadStartTime > windowStart) {
+                blockedTime += threadStartTime - windowStart;
+            }
+            final double newFlushTime = threadMetrics.getOrDefault("flush-time-total", 0.0);
```

Comment:
I have not seen the change for adding this metric, but just assuming this is measuring the time we call `store.flush` here.